lib/engine/ability/reservation.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Ability
+    class Reservation < Base
+      attr_accessor :city, :tile
+      attr_reader :hex, :slot
+
+      def setup(hex:, city: 0, slot: 0)
```

Comment:
it's better if city and slot are default nil and then city slot are filled on lines 13 14 with ||

Reply:
7abca3c1